torch/_dynamo/output_graph.py

```diff
@@ -1829,6 +1842,19 @@ def maybe_lift_tracked_freevar_to_input(self, arg):
             return arg
         return self.lift_tracked_freevar_to_input(arg)
 
+    def track_mutations_for_tensor_inputs(self, proxies: List[fx.Proxy]):
+        self.tensor_input_versions.update(
```

Comment:
The version counter logic gets disabled when torch.inference_mode is active. I don't think this gets used often, but I could be wrong. Could we:
- not track this information when torch.inference_mode is active
- raise unimplemented in cond if torch.inference_mode is active

Reply:
Actually, I'm not sure that torch.inference_mode isn't used often.

Reply:
Proposal #2, that doesn't use version counters:
- once we have captured a graph, we run clones of the input FakeTensors through the graph with a torch_dispatch mode active
- the torch_dispatch mode will raise an error if any inputs are being mutated. We can tell this by looking at the schema of the operators.

Downsides: doesn't really work with subclasses